test_conformance/api/test_queries.cpp

```diff
@@ -656,6 +656,57 @@ REGISTER_TEST(get_device_info)
     return 0;
 }
 
+REGISTER_TEST(get_device_info_comparability)
+{
+    int error = CL_SUCCESS;
+
+    // comparability test for CL_DEVICE_PLATFORM
+    // 1. find platform related to device without using query
+    cl_uint num_platforms = 0;
+    error = clGetPlatformIDs(16, nullptr, &num_platforms);
+    test_error(error, "clGetPlatformIDs failed");
+
+    std::vector<cl_platform_id> platforms(num_platforms);
+
+    error = clGetPlatformIDs(num_platforms, platforms.data(), &num_platforms);
+    test_error(error, "clGetPlatformIDs failed");
+
+    cl_uint num_devices = 0;
+    cl_platform_id comp_platform = nullptr;
+    for (int p = 0; p < (int)num_platforms && comp_platform == nullptr; p++)
+    {
+        error = clGetDeviceIDs(platforms[p], CL_DEVICE_TYPE_ALL, 0, nullptr,
+                               &num_devices);
+        test_error(error, "clGetDeviceIDs failed");
+
+        std::vector<cl_device_id> devices(num_devices);
+        error = clGetDeviceIDs(platforms[p], CL_DEVICE_TYPE_ALL, num_devices,
+                               devices.data(), nullptr);
+        test_error(error, "clGetDeviceIDs failed");
+
+        // try to find invalid device
```

Comment:
Nitpick: I don't think this is an "invalid device" that we are looking for.

Reply:
Done